8/s2i/bin/assemble

```diff
@@ -28,7 +28,7 @@ safeLogging () {
 
 shopt -s dotglob
 echo "---> Installing application source ..."
-mv /tmp/src/* ./
+rsync -av /tmp/src/ .
```

Comment:
this will leave the source in the injected location which results in two copies of it in the output image, no? we don't want that.

Reply:
Also this mv pattern is in pretty much all our s2i builder images, so if we do decide to make a change, all of them should be updated.

Reply:
> this will leave the source in the injected location which results in two copies of it in the output image, no? we don't want that.

yes, we could also `rm -rf /tmp/src/*` after, so image contains source repo only once. And also this merges into the files previous s2i image in the "chain" left in `./`. 

Or maybe we would like to plain `rm -rf ./* && mv /tmp/src* ./` to get rid of anything the previous image in the "chain" left there?

> Also this mv pattern is in pretty much all our s2i builder images, so if we do decide to make a change, all of them should be updated.

absolutely, marking this PR as [WIP] to serve a purpose of initiating a discussion about how to solve this for all s2i images

Reply:
We do want it to be possible to perform incremental builds with these images, so i do agree it needs to be solved.

As for which approach to take (clearing the destination first and doing a mv, or using rsync and then clearing the source), i don't have an informed opinion right now.

Reply:
> rm -rf ./* && mv /tmp/src* ./

Is the same as to prohibit using image "in such build chain", or not?

I think if we want to support similar chains, rsync is the only option. Because if user have any non-empty directory in the source, current implementation of assemble script will fail.

Reply:
> Is the same as to prohibit using image "in such build chain", or not?

yes, very similar. Maybe more than prohibiting I'd say ignoring, but either way the result ends up similar

> I think if we want to support similar chains, rsync is the only option

Thanks, I will update this PR for all assemble scripts (currently, it is only for nodejs) and wait for more feedback